commit/merkleroot/outcome.go

```diff
@@ -219,48 +219,99 @@ func buildMerkleRootsOutcome(
 }
 
 // filterRootsBasedOnRmnSigs filters the roots to only include the ones that are either:
-// 1) RMN-enabled and have RMN signatures
-// 2) RMN-disabled and do not have RMN signatures
+// 1) RMN-enabled and roots exist for all the roots that are included in the RMN signature.
+// 2) RMN-disabled and we do not have RMN signatures
 func filterRootsBasedOnRmnSigs(
 	lggr logger.Logger,
 	signedLaneUpdates []*rmnpb.FixedDestLaneUpdate,
-	roots []cciptypes.MerkleRootChain,
+	proposedRoots []cciptypes.MerkleRootChain,
 	rmnEnabledChains map[cciptypes.ChainSelector]bool,
 	addressCodec cciptypes.AddressCodec,
 ) []cciptypes.MerkleRootChain {
-	// Create a set of signed roots for quick lookup.
+	signedRoots, err := computeSignedRootsSet(lggr, signedLaneUpdates, addressCodec)
+	if err != nil {
+		lggr.Errorw("failed to compute signed roots set, skipping RMN-enabled roots", "err", err)
+		// we don't return since we still want to make progress with the non-RMN related roots (RMN disabled chains)
+	}
+	lggr.Debugw("computed signed roots set", "signedRoots", signedRoots.ToSlice())
+
+	// If at least ONE root that is signed is not proposed
+	// then we cannot make progress with ANY of the existing signed roots
+	// since the signature will be invalid
+	proposedRootsSet := mapset.NewSet[rootKey]()
+	for _, root := range proposedRoots {
+		addrStr, err := addressCodec.AddressBytesToString(root.OnRampAddress, root.ChainSel)
+		if err != nil {
+			lggr.Errorw("convert proposed root OnRamp address to string to build a set", "root", root, "err", err)
+			continue
+		}
+		proposedRootsSet.Add(rootKey{
+			ChainSel:      root.ChainSel,
+			SeqNumsRange:  root.SeqNumsRange,
+			MerkleRoot:    root.MerkleRoot,
+			OnRampAddress: addrStr,
+		})
+	}
+	if !signedRoots.IsSubset(proposedRootsSet) {
+		lggr.Errorw("signed roots are not a subset of proposed roots, skipping RMN-enabled roots",
```

Comment:
Just for my understanding: Is this indicative of a bigger problem? i.e. RMN does not compute the same root as the commit plugin?